libsolidity/ast/AST.h

```diff
@@ -869,7 +869,7 @@ class FunctionDefinition: public CallableDeclaration, public StructurallyDocumen
 			(annotation().contract && annotation().contract->isInterface());
 	}
 
-	FunctionDefinition const& resolveVirtual(
+	[[nodiscard]] FunctionDefinition const& resolveVirtual(
```

Comment:
It is a const function with const ref / pointer parameters. What is the purpose of the ``[[nodiscard]]`` decorator?

Reply:
The purpose is to not call it and then forget to use the result.. which is exactly what happend to me

Reply:
In that case, we have to add `[[nodiscard]]` to all functions that return something.

Reply:
I think it really clutters the code a lot. I mean if you return something that releases a resource upon destruction and because of that should really be used, then `nodiscard' makes sense, but I really don't understand why a function returning a const ref should benefit from this tag.